torch/csrc/jit/tensorexpr/tensor.h

```diff
@@ -12,7 +12,7 @@ namespace torch {
 namespace jit {
 namespace tensorexpr {
 
-class Tensor {
+class Tensor : KernelScopedObject {
```

Comment:
Please manually checked that there is no leaked object, or double released ones. This looks good in general, I am a bit concerned that nothing has caught this yet.

Reply:
I'm pretty sure we leaked before, as we never freed our tensors. Share your concern about not catching it earlier, I'm not sure why it doesn't show up in asan testing in our "enable-by-default" PR.